src/main/java/buchen/station/CitiBikeController.java

```diff
@@ -0,0 +1,154 @@
+package buchen.station;
+
+import org.jxmapviewer.JXMapViewer;
+import org.jxmapviewer.OSMTileFactoryInfo;
+import org.jxmapviewer.input.CenterMapListener;
+import org.jxmapviewer.input.PanKeyListener;
+import org.jxmapviewer.input.PanMouseInputListener;
+import org.jxmapviewer.input.ZoomMouseWheelListenerCursor;
+import org.jxmapviewer.painter.CompoundPainter;
+import org.jxmapviewer.viewer.*;
+import org.jxmapviewer.painter.Painter;
+
+import javax.swing.event.MouseInputListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Point2D;
+import java.util.*;
+import java.util.function.BiConsumer;
+
+public class CitiBikeController {
+    private JXMapViewer mapViewer;
+    boolean start = false;
+    private GeoPosition startLocation;
+    private GeoPosition endLocation;
+    private Station startStation;
+    private Station endStation;
+    private BiConsumer<Double, Double> startPointDouble;
+    private BiConsumer<Double, Double> endPointDouble;
+    private final List<GeoPosition> track = new ArrayList<>();
+
+    public JXMapViewer getMap() {
+        mapViewer = new JXMapViewer();
+
+        TileFactoryInfo info = new OSMTileFactoryInfo();
+        DefaultTileFactory tileFactory = new DefaultTileFactory(info);
+        mapViewer.setTileFactory(tileFactory);
+
+        tileFactory.setThreadPoolSize(8);
+
+        mapViewer.setZoom(7);
+        GeoPosition nyc = new GeoPosition(40.77228687788679, -73.9842939376831);
+        mapViewer.setAddressLocation(nyc);
+
+        MouseInputListener mia = new PanMouseInputListener(mapViewer);
+        mapViewer.addMouseListener(mia);
+        mapViewer.addMouseMotionListener(mia);
+        mapViewer.addMouseListener(new CenterMapListener(mapViewer));
+        mapViewer.addMouseWheelListener(new ZoomMouseWheelListenerCursor(mapViewer));
+        mapViewer.addKeyListener(new PanKeyListener(mapViewer));
+
+
+        mapViewer.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                int x = e.getX();
+                int y = e.getY();
+                Point2D.Double point = new Point2D.Double(x, y);
+                    if (!start) {
+                        startLocation = mapViewer.convertPointToGeoPosition(point);
+                        startPointDouble.accept(startLocation.getLatitude(), startLocation.getLongitude());
+                        start = true;
+                    } else
+                    {
+                        endLocation = mapViewer.convertPointToGeoPosition(point);
+                        endPointDouble.accept(endLocation.getLatitude(), endLocation.getLongitude());
+                    }
+                updateWaypoints();
+            }
+        });
+        return mapViewer;
+    }
+
+    public void showRoute() {
+        CitiBikeServiceFactory factory = new CitiBikeServiceFactory();
```

Comment:
You are supposed to use your lamda, running in the cloud, to do this. Not access the CitiBike API directly.